src/blocks.zig

```diff
@@ -374,11 +378,41 @@ pub const Block = packed struct { // MARK: Block
 		return _allowOres[self.typ];
 	}
 
+	pub inline fn touchFunction(self: Block) ?*const TouchFunction {
+		return _touchFunction[self.typ];
+	}
+
 	pub fn canBeChangedInto(self: Block, newBlock: Block, item: main.items.ItemStack, shouldDropSourceBlockOnSuccess: *bool) main.rotation.RotationMode.CanBeChangedInto {
 		return newBlock.mode().canBeChangedInto(self, newBlock, item, shouldDropSourceBlockOnSuccess);
 	}
 };
 
+pub const TouchFunction = fn(block: Block, entity: Entity, posX: i32, posY: i32, posZ: i32, isEntityInside: bool) void;
+
+pub const TouchFunctions = struct {
+	var hashMap: std.StringHashMap(*const TouchFunction) = undefined;
+
+	pub fn init() void {
+		hashMap = .init(main.globalAllocator.allocator);
+		inline for(@typeInfo(TouchFunctions).@"struct".decls) |declaration| {
+			if(@TypeOf(@field(TouchFunctions, declaration.name)) == TouchFunction) {
+				hashMap.putNoClobber(declaration.name, &@field(TouchFunctions, declaration.name)) catch unreachable;
+			}
+		}
+	}
+
+	pub fn deinit() void {
+		hashMap.deinit();
+	}
+
+	pub fn getFunctionPointer(id: []const u8) ?*const TouchFunction {
+		const pointer = hashMap.getPtr(id);
+		if(pointer == null)
+			return null;
```

Comment:
Please print an error if the string wasn't empty, so addon creators can discover if they made a typo or something.